core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -879,70 +874,60 @@ object IO {
     }
 
   /**
-   * Apply the function fn to each element of the `TraversableOnce[A]` and
-   * return the results in a new `TraversableOnce[B]`. For parallelism use `parTraverse`.
+   * Apply the function fn to each element of the `List[A]` and
+   * return the results in a new `List[B]`. For parallelism use `parTraverse`.
    */
-  final def traverse[E, A, B, M[X] <: TraversableOnce[X]](
-    in: M[A]
-  )(fn: A => IO[E, B])(implicit cbf: CanBuildFrom[M[A], B, M[B]]): IO[E, M[B]] =
-    in.foldLeft[IO[E, mutable.Builder[B, M[B]]]](IO.sync(cbf(in)))((io, b) => io.zipWith(fn(b))(_ += _))
-      .map(_.result())
+  final def traverse[E, A, B](in: List[A])(fn: A => IO[E, B]): IO[E, List[B]] =
```

Comment:
Returning a `List` seems sensible, but there is no cost to accept a `Iterable`, which exists < 2.12 and >= 2.13.